airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -169,8 +178,17 @@ def read(self, source_spec: ConnectorSpecification, config: TConfig, catalog: An
 
         # The Airbyte protocol dictates that counts be expressed as float/double to better protect against integer overflows
         stream_message_counter: DefaultDict[HashableStreamDescriptor, float] = defaultdict(float)
-        for message in self.source.read(self.logger, config, catalog, state):
-            yield self.handle_record_counts(message, stream_message_counter)
+        try:
```

Comment:
nit: can you separate this change since it's not directly related to to the discover error handling?

Reply:
@girarda 
 TLDR: `read` for file-based sources can invoke `get_json_schema` which I updated from wrapping `SchemaInferenceError` in an `AirbyteTracedException` to just raising it so it can get caught by the new error handling -- so I moved wrapping it to `entrypoint` `read`. Let me know if you don't like this approach.

Longer version:
`entrypoint` `read` invokes the `FileBasedSource` `streams` method which *can* (if concurrency_level is not `None) result in initializing the `FileBasedStreamFacade` which in turns initializes `DefaultStream` with a json_schema argument set to `stream.get_json_schema()` which no longer wraps the `SchemaInferenceError` in an ATE.

Reply:
After discussion, we've decided to revert this and more specifically, the way `InvalidSchemaError` is handled within `default_file_based_stream` `get_json_schema` method. It now gets wrapped in an `AirbyteTracedException` like before, and the `file_based_discover_error_handler` now checks for the underlying exception if an `AirbyeTracedException` is being handled.